src/containers/pages/JurisdictionAssignment/AutoSelectJurisdictions/index.tsx

```diff
@@ -0,0 +1,265 @@
+    const metaDataPromise = loadJurisdictionsMetadata(
+      JURISDICTION_METADATA_RISK,
+      OpenSRPService,
+      fetchJurisdictionsMetadataCreator
+    ).catch(e => handleBrokenPage(e.message));
+
+    // create promise to fetch the plan
+    const planId = props.match.params.planId;
+    const planDataPromise = loadOpenSRPPlan(planId, serviceClass, fetchPlanCreator).catch(_ => {
+      handleBrokenPage(COULD_NOT_LOAD_PLAN);
+    });
+
+    Promise.all([metaDataPromise, planDataPromise])
+      .finally(() => {
+        stopLoading(planAndMetaLoadingKey);
+      })
+      .catch(e => handleBrokenPage(e.message));
+  }, []);
+
+  const rootJurisdictionId = useGetRootJurisdictionId(
+    plan,
+    serviceClass,
+    handleBrokenPage,
+    stopLoading,
+    startLoading
+  );
+
+  useGetJurisdictionTree(
+    rootJurisdictionId,
+    startLoading,
+    treeFetchedCreator,
+    stopLoading,
+    handleBrokenPage
+  );
+
+  if (loading()) {
+    return <Ripple />;
+  }
+  if (!plan) {
+    return <ErrorPage errorMessage={COULD_NOT_LOAD_PLAN} />;
+  }
+
+  if (!rootJurisdictionId) {
+    return <ErrorPage errorMessage={COULD_NOT_LOAD_JURISDICTION} />;
+  }
+  if (broken) {
+    return <ErrorPage errorMessage={errorMessage} />;
+  }
+
+  const sliderProps = {
+    onClickNext: () => setStep(step + 1),
+    plan,
+    rootJurisdictionId,
+    serviceClass,
+  };
+
+  const jurisdictionTableProps = {
+    // tree,
+    autoSelectionFlow: true,
+    currentParentId: props.match.params.parentId,
+    jurisdictionsMetadata,
+    plan,
+    rootJurisdictionId,
+    serviceClass,
+  };
+
+  const structureTableProps = {
+    ...props,
+    currentParentId: props.match.params.parentId,
+    onClickNext: () => setStep(step + 1),
+    rootJurisdictionId,
+  };
+
+  const breadcrumbProps = {
+    currentPage: {
+      label: plan.title,
+      url: `${AUTO_ASSIGN_JURISDICTIONS_URL}/${plan.identifier}`,
+    },
+    pages: [
+      {
+        label: PLANNING_PAGE_TITLE,
+        url: PLANNING_VIEW_URL,
+      },
+    ],
+  };
+
+  const timelineSliderProps: TimelineSliderProps = {
+    keyOfCurrentStop: step.toString(),
+    stops: [
+      {
+        keys: '1',
+        labelBelowStop: AUTO_TARGET_JURISDICTIONS_BY_RISK,
+        labelInStop: '1',
+      },
+      {
+        keys: ['2', '3'],
+        labelBelowStop: REFINE_SELECTED_JURISDICTIONS,
+        labelInStop: '2',
+      },
+    ],
+  };
+
+  const pageTitle = plan.title;
+  return (
+    <>
+      <Helmet>
+        <title>{pageTitle}</title>
+      </Helmet>
+      <HeaderBreadcrumb {...breadcrumbProps} />
+      <h3 className="mb-3 page-title">{pageTitle}</h3>
+      <TimelineSlider {...timelineSliderProps} />
+      {step === 1 && <ConnectedJurisdictionSelectionsSlider {...sliderProps} />}
+      {step === 2 && <ConnectedSelectedStructuresTable {...structureTableProps} />}
+      {step === 3 && <ConnectedJurisdictionTable {...jurisdictionTableProps} />}
```

Comment:
- Lets do something about the magic numbers here
- Add a comment to explain this

Reply:
done